android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/RegisterRepository.kt

```diff
@@ -113,8 +103,8 @@ constructor(
 
     // Retrieve data for each of the configured related resources
     // Also retrieve data for nested related resources for each of the related resource
-    baseResources.map { baseResource: Resource ->
-      val currentRelatedResources = LinkedList<RelatedResourceData>()
+    return baseResources.map { baseResource: Resource ->
+      val currentRelatedResources = LinkedList<RepositoryResourceData>()
 
       relatedResourcesConfig.forEach { resourceConfig: ResourceConfig ->
```

Comment:
is the `searchRelatedResources` expensive? This might be another loop to parallelize

Reply:
this is already parallelized

Reply:
my bad, I missed that